Changes to add the specification extensions SUP content (#190)
margo/specification: system-design/specification/margo-management-interface/specification-extensions.md

Workload fleet management server and client implementations MUST NOT fail when encountering unknown specification extensions in the Application Description, Application Deployment or Device Capabilities documents. Unknown specification extension properties targeting a workload fleet manager or device supplier MUST be ignored.

For specifics, see the specification extension documentation for each of the following documents:

- [Application Description](../applications/application-description.md#specification-extensions)
- [Application Deployment](./desired-state#specification-extensions)
- [Device Capabilities](./device-capabilities.md#specification-extensions)

The property naming conventions for adding specification extension is `x-<unique name>-extensions`. 

The `<unique-name>` MUST match the `[a-z][a-z0-9-]*` regular expression and conform with the following rules:

- MUST start with a lowercase letter
- MUST have no spaces or special characters 

> - MAY container lowercase letters, digits, and dashes

Reviewer: ```suggestion
- MAY contain lowercase letters, digits, and dashes
```

> It is recommend to use identifying information as the unique name such as a company name, organization name, a combination of company name and department, etc. to reduce chances of naming conflicts.

Reviewer: ```suggestion
It is recommended to use identifying information as the unique name such as a company name, organization name, a combination of company name and department, etc. to reduce chances of naming conflicts.
```

For example:

- x-acme-extensions
- x-edgecircuitsystems-extensions
- x-northstar-industries-extensions
- x-northforgesystems-processautomation-extensions

> Extension properties may contain any simple or complex child properties as long as it is valid YAML or JSON applicable to the document type the specification extension property is added to.

Reviewer: The schema in the OpenAPI spec limits extensions to being JSON objects (`type: object`). I think this is a valid choice, but the prose doesn't match that.

```suggestion
The value of an extension property MUST be an object (a YAML/JSON mapping). Its child properties may be any simple or complex structure, as long as it is valid YAML or JSON applicable to the document type the specification extension property is added to.
```
